### PR TITLE
[CEN-1440] Update AppInsights to latest release

### PR DESCRIPTION
**Pull request by alberto-re-pagopa**
#### Description

This PR updates the Java AppInsights agent to the newest release.
It also adopt the best practice of multi-stage build in the Dockerfile, thus ensuring that the build environment is also defined as-a-code.

#### List of Changes

- Updated AppInsights to newest release (3.2.7 as today)
- Refactored Dockerfile as a multi-stage build

#### Motivation and Context

- We want to enable AppInsights monitoring on the rtd-ms-microservice

#### How Has This Been Tested?

- [ ] Unit Test Suite
- [ ] Integration Test Suite
- [x] Api Tests
- [ ] Load Tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by TommasoLencioni — COMMENTED** [review]
Looks good.

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,16 @@
-FROM adoptopenjdk/openjdk11:alpine-jre
-ADD https://github.com/microsoft/ApplicationInsights-Java/releases/download/3.2.6/applicationinsights-agent-3.2.6.jar /applicationinsights-agent.jar
+FROM maven:3.8.4-jdk-11-slim as buildtime
+
+WORKDIR /build
+COPY . .
+
+RUN mvn clean package
```
Comment (GiovanniMancini) [inline Dockerfile:6]: @alberto-re-pagopa Using a multistage build may be a cleaner solution, but we're already building the project in the deploy pipeline. As a result, we're build it two times

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+
 VOLUME /tmp
-COPY target/*.jar app.jar
-ENTRYPOINT ["java","-jar","app.jar"]
\ No newline at end of file
+WORKDIR /app
+
+COPY --from=buildtime /build/target/*.jar /app/app.jar
+ADD https://github.com/microsoft/ApplicationInsights-Java/releases/download/3.2.7/applicationinsights-agent-3.2.7.jar /app/applicationinsights-agent.jar
+
```
Comment (GiovanniMancini) [inline Dockerfile:15]: I'll add a comment to explain that we are setting the app insight agent via  `JAVA_TOOL_OPTIONS`

**Review by GiovanniMancini — APPROVED** [review]
We have to choice if the build is done via a multistage docker file or via a DevOps task

File: Dockerfile (hunk 2 above)
Comment (alberto-re-pagopa) [reply Dockerfile:15]: Agree, could save some time in the future

**Pull request merged**